File: src-tauri/src/models/link.rs
use serde::{Deserialize, Serialize};

#[derive(Serialize, Deserialize)]
pub struct Link {
    pub id: i32,
    pub slug_title: String,
    pub title: String,
    pub description: Option<String>,
    pub thumbnail_image: Option<String>,
    pub link_id: i32,
    pub link_type: String,
}

#[derive(Serialize, Deserialize)]
pub struct LinkId {
    pub id: i32,
    pub from_memo_id: i32,
    pub to_memo_id: i32,
}

File: src/models/link.ts
export type Link = {
  id: number;
  slug_title: string;
  title: string;
  description: string | null;
  thumbnail_image: string | null;
  link_id: number;
  link_type: string;
};

export type BaseLink = {
  id: number | string;
  slug_title: string;
  title: string;
};

export type ExistingLink = BaseLink & {
  tag: 'existing';
};

export type NewLink = BaseLink & {
  tag: 'new';
};

export type LinkPaletteItem = ExistingLink | NewLink;

export type LinkPaletteItems = Array<{
  key: string;
  label: string;
  commands: Array<LinkPaletteItem>;
}>;
